### PR TITLE
Verify encryption signature

### PR DESCRIPTION
**Pull request by snormore**
There's a bug in the interaction with LedgerLive for iOS where the previous signature is returned as the response of a consecutive signature request, and so this PR verifies that signature, and retries if it's not what's expected, which seems to solve the issue when interacting with LedgerLive.

### CODE REVIEW

**Comment by mkobetic** [comment]
> The PR also updates the decryption logic to recreate the improperly encrypted key bundle if it fails to decrypt, so that existing accounts that may have encountered this bug can continue through their currently broken initialization workflow.

This bit worries, are we sure that by recreating the bundle we won't lose previously used keys? That may be similarly catastrophic for a user.

File: src/store/EncryptedStore.ts [hunk 1]
```diff
@@ -73,9 +73,31 @@ export default class EncryptedKeyStore implements KeyStore {
...
+    let address = utils.recoverAddress(utils.hashMessage(input), sig)
+    if (address !== walletAddr) {
+      sig = await wallet.signMessage(input)
+      console.log('invalid signature, retrying', address, walletAddr)
+    }
+
+    address = utils.recoverAddress(utils.hashMessage(input), sig)
+    if (address !== walletAddr) {
+      throw new Error('invalid signature')
+    }
```
Comment (jazzz) [inline src/store/EncryptedStore.ts:94]: In this case I don't think lower level recoverAddress is needed. Verify message should work

```
const recoveredAddr = utils.verifyMessage(input, sig)
if ((await wallet.getAddress()) === recoveredAddr) {
  throw new Error('invalid storage signature')
}
```

**Review by jazzz — APPROVED** [review]

File: src/store/EncryptedStore.ts [hunk 2]
```diff
@@ -73,9 +73,27 @@ export default class EncryptedKeyStore implements KeyStore {
...
+
+    // Check that the signature is correct, was created using the expected
+    // input, and retry if not. This mitigates a bug in interacting with
+    // LedgerLive for iOS, where the previous signature response is
+    // returned in some cases.
+    let address = utils.verifyMessage(input, sig)
+    if (address !== walletAddr) {
+      sig = await wallet.signMessage(input)
+      console.log('invalid signature, retrying')
+    }
```
Comment (mkobetic) [inline src/store/EncryptedStore.ts:89]: Nice, not for this PR, but we should probably use `verifyMessage()` in `WalletSigner.signerKey()` as well to simplify the code there.

**Review by mkobetic — APPROVED** [review]

**Comment by snormore** [comment]
Does this need to be merged into the `beta` branch too?

**Comment by neekolas** [comment]
> Does this need to be merged into the `beta` branch too?

Yes please

**Pull request merged**